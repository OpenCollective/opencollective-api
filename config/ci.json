{
  "database": {
    "url": "postgres://opencollective@localhost:5432/opencollective_dvl",
    "options": {
      "logging": false,
      "benchmark": true
    }
  },
  "maintenancedb": {
    "url": "postgres://postgres@localhost:5432/postgres"
  },
  "limits": {
    "virtualCards": {
      "maxPerDay": 1000
    }
  },
  "redis": {
    "serverUrl": "redis://localhost"
  },
  "log": {
    "level": "warn"
  },
  "aws": {
    "s3": {
      "bucket": "opencollective-test"
    }
  },
  "mailgun": {
    "user": "",
    "password": ""
  },
  "fixer": {
    "accessKey": "UNDEFINED_FIXER_ACCESS_KEY"
  },
  "twoFactorAuth": {
    "secretKey": "guineapigs",
    "cipher": "DES"
  }
}
