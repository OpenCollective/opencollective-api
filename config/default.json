{
  "database": {
    "database": "opencollective_localhost",
    "username": "opencollective",
    "password": "",
    "options": {
      "host": "127.0.0.1",
      "dialect": "postgres",
      "logging": false
    }
  },
  "keys": {
    "opencollective": {
      // TODO: This needs to be changed.
      "resetPasswordSecret": "O4ExDVzM7OviRzQBhRSV0Z2P1w4impwH"
    }
  },
  "host": {
    "api": "http://localhost:3060",
    "webapp": "http://localhost:3000/app",
    "website": "http://localhost:3000"
  },
  "email": {
    "from": "OpenCollective <info@opencollective.com>"
  },
  "mailgun": {
    "user": "",
    "password": ""
  },
  "paypal": {
    "classic": {
      "email": "",
      "appId": "",
      "userId": "",
      "password": "",
      "signature": ""
    },
    "rest": {
      "mode": "sandbox"
    },
    "feeOC": 5
  },
  "stripe": {
    "redirectUri": "http://localhost:3060/stripe/oauth/callback"
  },
  "clearbit": "",
  "slack": {
    "hookUrl": "",
    "activityChannel": "#activity_test"
  }
}
