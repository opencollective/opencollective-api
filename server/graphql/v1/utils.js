import debugLib from 'debug';
import { graphql } from 'graphql';

import { loaders } from '../loaders';

import schema from './schema';

const debug = debugLib('graphql');

export const makeRequest = (remoteUser, query) => {
  return {
    remoteUser,
    body: { query },
    loaders: loaders({ remoteUser }),
  };
};

export const graphqlQuery = async (query, variables, remoteUser) => {
  const prepare = () => {
    if (remoteUser) {
      remoteUser.rolesByCollectiveId = null; // force refetching the roles
      return remoteUser.populateRoles();
    } else {
      return Promise.resolve();
    }
  };

  if (process.env.DEBUG && process.env.DEBUG.match(/graphql/)) {
    debug('query', query);
    debug('variables', variables);
    debug('context', remoteUser);
  }

  return prepare().then(() =>
    graphql({
      schema,
      source: query,
      rootValue: null,
      contextValue: makeRequest(remoteUser, query),
      variableValues: variables,
    }),
  );
};
