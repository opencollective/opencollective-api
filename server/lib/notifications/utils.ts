import sanitizeHtml from 'sanitize-html';

import { sanitizerOptions as updateSanitizerOptions } from '../../models/Update';

const constructPreviewImageURL = (service: string, id: string) => {
  if (service === 'youtube' && id.match('[a-zA-Z0-9_-]{11}')) {
    return `https://img.youtube.com/vi/${id}/0.jpg`;
  } else if (service === 'anchorFm') {
    return `https://opencollective.com/static/images/anchor-fm-logo.png`;
  } else {
    return null;
  }
};

const parseServiceLink = (videoLink: string) => {
  const regexps = {
    youtube: new RegExp(
      '(?:https?://)?(?:www\\.)?youtu(?:\\.be/|be(-nocookie)?\\.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
      'i',
    ),
    anchorFm: /^(http|https)?:\/\/(www\.)?anchor\.fm\/([^/]+)(\/embed)?(\/episodes\/)?([^/]+)?\/?$/,
  };
  for (const service in regexps) {
    videoLink = videoLink.replace('/?showinfo=0', '');
    const matches = regexps[service].exec(videoLink);
    if (matches) {
      if (service === 'anchorFm') {
        const podcastName = matches[3];
        const episodeId = matches[6];
        const podcastUrl = `${podcastName}/embed`;
        return { service, id: episodeId ? `${podcastUrl}/episodes/${episodeId}` : podcastUrl };
      } else {
        return { service, id: matches[matches.length - 1] };
      }
    }
  }
  return {};
};

export const replaceVideosByImagePreviews = (html: string) => {
  const sanitizerOptions = {
    ...updateSanitizerOptions,
    transformTags: {
      ...updateSanitizerOptions.transformTags,
      iframe: (tagName, attribs) => {
        if (!attribs.src) {
          return '';
        }
        const { service, id } = parseServiceLink(attribs.src);
        const imgSrc = constructPreviewImageURL(service, id);
        if (imgSrc) {
          return {
            tagName: 'img',
            attribs: {
              src: imgSrc,
              alt: `${service} content`,
            },
          };
        } else {
          return '';
        }
      },
    },
  };
  return sanitizeHtml(html, sanitizerOptions);
};
