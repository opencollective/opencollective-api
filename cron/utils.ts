import config from 'config';

/**
 * Heroku scheduler only has daily or hourly cron jobs, we only want to run
 * this script once per week on Monday (1). If the day is not Monday on production
 * we won't execute the script
 */
export function onlyExecuteInProdOnMondays() {
  const today = new Date();
  if (config.env === 'production' && today.getDay() !== 1) {
    console.log('OC_ENV is production and day is not Monday, script aborted!');
    process.exit(0);
  }
}
