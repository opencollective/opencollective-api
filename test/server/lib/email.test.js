import { expect } from 'chai';
import config from 'config';
import nodemailer from 'nodemailer';
import { stub } from 'sinon';

import emailLib from '../../../server/lib/email';
import { md5 } from '../../../server/lib/utils';
import * as utils from '../../utils';

const emailData = utils.data('emailData');

describe('server/lib/email', () => {
  describe('Sending emails', () => {
    let nm;

    // create a fake nodemailer transport
    beforeEach(done => {
      config.mailgun.user = 'xxxxx';
      config.mailgun.password = 'password';

      nm = nodemailer.createTransport({
        name: 'testsend',
        service: 'Mailgun',
        sendMail(data, callback) {
          callback();
        },
        logger: false,
      });
      stub(nodemailer, 'createTransport').callsFake(() => {
        return nm;
      });
      done();
    });

    // stub the transport
    beforeEach(done => {
      stub(nm, 'sendMail').callsFake((object, cb) => {
        cb(null, object);
      });
      done();
    });

    afterEach(done => {
      nm.sendMail.restore();
      done();
    });

    afterEach(() => {
      config.mailgun.user = '';
      config.mailgun.password = '';
      nodemailer.createTransport.restore();
    });

    it('sends the thankyou.fr email template', async () => {
      const template = 'order.thankyou';
      const collective = { name: 'En Marche', slug: 'enmarchebe' };
      const data = {
        order: { totalAmount: 5000, currency: 'EUR' },
        transaction: { uuid: '17811b3e-0ac4-4101-81d4-86e9e0aefd7b' },
        config: { host: config.host },
        interval: 'month',
        firstPayment: false,
        user: emailData.user,
        fromCollective: { id: 1, slug: 'xdamman', name: 'Xavier' },
        collective,
      };
      const options = {
        from: `"${collective.name}" <hello@opencollective.com>`,
      };
      await emailLib.send(template, data.user.email, data, options);
      let amountStr = 50;
      amountStr = amountStr.toLocaleString('fr-FR', {
        style: 'currency',
        currency: 'EUR',
        minimumFractionDigits: 2,
        maximumFractionDigits: 2,
      });
      expect(nm.sendMail.lastCall.args[0].from).to.equal(options.from);
      expect(nm.sendMail.lastCall.args[0].to).to.equal('emailbcc+user1-at-opencollective.com@opencollective.com');
      expect(nm.sendMail.lastCall.args[0].subject).to.contain(
        `Merci pour votre contribution de ${amountStr}/mois à En Marche`,
      );
      expect(nm.sendMail.lastCall.args[0].html).to.contain('Merci pour continuer à nous soutenir');
      expect(nm.sendMail.lastCall.args[0].html).to.contain('donate');
      expect(nm.sendMail.lastCall.args[0].headers['X-Mailgun-Tag']).to.equal('internal');

      expect(nm.sendMail.lastCall.args[0].html).to.matchSnapshot();
      expect(nm.sendMail.lastCall.args[0].text).to.matchSnapshot();

      expect(nm.sendMail.lastCall.args[0].headers).to.containSubset({
        'X-OpenCollective-Account': 'enmarchebe',
        'List-ID': 'enmarchebe::order.thankyou',
        'List-Unsubscribe-Post': 'List-Unsubscribe=One-Click',
        'List-Unsubscribe':
          '<http://localhost:3000/email/unsubscribe/user1%40opencollective.com/enmarchebe/order.thankyou/54349853da7f4493e08265f3a6d600a9b705aec3bea42cb8ce23688c3060c3837d04a6285263edfdcdc83cfd7a7f95692f274a1c4d7f40f4c0db031e638589e9>',
      });
    });

    it('sends the thankyou.wwcode email template', async () => {
      const paymentData = {
        totalAmount: 5000,
        currency: 'USD',
      };

      const data = {
        order: paymentData,
        transaction: { uuid: '17811b3e-0ac4-4101-81d4-86e9e0aefd7b' },
        config: { host: config.host },
        interval: 'month',
        user: emailData.user,
        collective: {
          name: 'WWCode Austin',
          slug: 'wwcodeaustin',
        },
      };
      await emailLib.send('order.thankyou', data.user.email, data);
      let amountStr = 50;
      amountStr = amountStr.toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD',
        minimumFractionDigits: 2,
        maximumFractionDigits: 2,
      });
      expect(nm.sendMail.lastCall.args[0].to).to.equal('emailbcc+user1-at-opencollective.com@opencollective.com');
      expect(nm.sendMail.lastCall.args[0].subject).to.contain(
        `Thank you for your ${amountStr}/month donation to WWCode Austin`,
      );

      expect(nm.sendMail.lastCall.args[0].headers).to.containSubset({
        'X-OpenCollective-Account': 'wwcodeaustin',
        'List-ID': 'wwcodeaustin::order.thankyou',
        'List-Unsubscribe-Post': 'List-Unsubscribe=One-Click',
        'List-Unsubscribe':
          '<http://localhost:3000/email/unsubscribe/user1%40opencollective.com/wwcodeaustin/order.thankyou/d33986f22d66bf7254173f354874ba1557b094f96ac12ebb0afbabfd9a3fe76ed1fc5c30662d7ecc318ff6a0a47f9b09b9e71797b72627423b9694710fd0c80e>',
      });
    });
  });

  describe('Unsubscribe', () => {
    const EMAIL_ADDRESS = 'user@opencollective.com';
    const COLLECTIVE_SLUG = 'collective_slug';
    const EMAIL_TYPE = 'test-notification';

    describe('generateUnsubscribeToken', () => {
      it('generates new tokens as SHA512', () => {
        const { generateUnsubscribeToken } = emailLib;
        const token = generateUnsubscribeToken(EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE);
        expect(token).to.eq(
          'c8b476c8ac6f5347fc059628947a61fb354125157ca9060de38cd1189b3257710c034e7d4d559787174c678a78de831737c8b6a631ec8a5a0a5edbd68a8a7e67',
        );
      });

      it('generates legacy tokens with MD5', () => {
        const { generateUnsubscribeToken } = emailLib;
        const token = generateUnsubscribeToken(EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE, md5);
        expect(token).to.eq('0227ed95b740e41d8359667e3e3baa9c');
      });
    });

    describe('isValidUnsubscribeToken', () => {
      it('detects invalid tokens', () => {
        const { isValidUnsubscribeToken } = emailLib;
        expect(isValidUnsubscribeToken('', EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE)).to.be.false;
        expect(isValidUnsubscribeToken('test', EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE)).to.be.false;
      });

      it('works with new tokens', () => {
        const { generateUnsubscribeToken, isValidUnsubscribeToken } = emailLib;
        const token = generateUnsubscribeToken(EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE);
        expect(isValidUnsubscribeToken(token, EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE)).to.be.true;
      });

      it('works with legacy tokens', () => {
        const { generateUnsubscribeToken, isValidUnsubscribeToken } = emailLib;
        const token = generateUnsubscribeToken(EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE, md5);
        expect(isValidUnsubscribeToken(token, EMAIL_ADDRESS, COLLECTIVE_SLUG, EMAIL_TYPE)).to.be.true;
      });
    });
  });

  describe('render', () => {
    it('Automatically generates text version', () => {
      const result = emailLib.render('user.new.token', { loginLink: 'https://opencollective/USER_LOGIN_LINK' });
      expect(result.html).to.matchSnapshot();
      expect(result.text).to.matchSnapshot();
    });
  });

  describe('generateFromEmailHeader', () => {
    it('sanitizes invalid characters', () => {
      expect(emailLib.generateFromEmailHeader('Hello World')).to.equal('"Hello World" <no-reply@opencollective.com>');
      expect(emailLib.generateFromEmailHeader('Hello World <NOPE>')).to.equal(
        '"Hello World <NOPE>" <no-reply@opencollective.com>',
      );
      expect(emailLib.generateFromEmailHeader('Hello "World" <NOPE>')).to.equal(
        '"Hello “World“ <NOPE>" <no-reply@opencollective.com>',
      );
      expect(
        emailLib.generateFromEmailHeader(
          `Hello
      
          “World“     <NOPE>`,
        ),
      ).to.equal('"Hello “World“ <NOPE>" <no-reply@opencollective.com>');
    });

    it('can customize the email address', () => {
      expect(emailLib.generateFromEmailHeader('Hello World', 'test@opencollective.com')).to.equal(
        '"Hello World" <test@opencollective.com>',
      );
    });
  });
});
