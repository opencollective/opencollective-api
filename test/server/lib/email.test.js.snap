// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server/lib/email Sending emails sends the thankyou.fr email template 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\">
<head>
  <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
  <title></title>
  <style>
@media only screen and (min-device-width: 601px) {
  .content {
    width: 600px !important;
  }
}
.btn.blue:hover {
  background: linear-gradient(180deg,#297EFF 0%,#1869F5 100%);
  background-color: #297EFF;
  border-color: #297EFF;
  color: #FFFFFF;
}
</style>
</head>
<body margin=\\"0\\" padding=\\"0\\" yahoo style=\\"margin: 0; padding: 0; min-width: 100%;\\">

<table bgcolor=\\"white\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\" style=\\"font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;box-sizing:border-box;font-size:14px;\\"><tr><td align=\\"center\\" valign=\\"top\\">
  <!--[if (gte mso 9)|(IE)]>
    <table width=\\"600\\" align=\\"center\\" cellpadding=\\"0\\" cellspacing=\\"0\\" border=\\"0\\"><tr><td>
  <![endif]-->
  <table class=\\"content\\" border=\\"0\\" width=\\"100%\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"width: 100%; max-width: 600px; padding: 0 10px;\\">
    <tr>
      <td></td>
      <td>


<p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">Xavier,</p>


<p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">Votre derni√®re contribution de 50,00¬†‚Ç¨ vient d'√™tre re√ßue. Merci pour continuer √† nous soutenir!</p>


<p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">Vous pouvez suivre nos prochaines activit√©s et comment on va utiliser l'argent recolt√© sur la page de notre collectif
  ouvert:
  <a href=\\"https://opencollective.com/enmarchebe\\" style=\\"text-decoration: none; color: #297EFF;\\">https://opencollective.com/enmarchebe</a>.
</p>

<h2 style=\\"color: #303233; font-family: 'Helvetica Neue'; font-weight: normal; font-size: 20px; line-height: 24px; text-align: center;\\">Aidez-nous √† r√©colter plus d'argent!</h2>
<p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">Partagez cette URL: <a href=\\"https://opencollective.com/enmarchebe/donate\\" style=\\"text-decoration: none; color: #297EFF;\\">https://opencollective.com/enmarchebe/donate</a>.





  </p><p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">Encore merci! üôè</p>

  <p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">
    ‚Äì Le collectif En Marche
  </p>

  </td>
  </tr>
  <tr>
    <td colspan=\\"3\\" height=\\"40\\"></td>
  </tr>
  <tr>
    <td colspan=\\"3\\" align=\\"center\\">
      <table width=\\"100%\\">
        <tr>
          <td></td>
          <td width=\\"200\\">
            <a href=\\"http://localhost:3000\\" style=\\"text-decoration: none; color: #297EFF;\\">
              <img width=\\"220\\" height=\\"28\\" src=\\"http://localhost:3000/static/images/email/logo-email-footer@2x.png\\" style=\\"max-width: 100%;\\">
            </a>
          </td>
          <td></td>
        </tr>
      </table>
    </td>
  </tr>
  <tr>
    <td colspan=\\"3\\" align=\\"center\\" style=\\"font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;box-sizing:border-box;font-size:12px;color:#999;\\">
      We can do great things together<br><br>
      To unsubscribe from thank you for your donation, <a href=\\"http://localhost:3000/email/unsubscribe/user1%40opencollective.com/enmarchebe/thankyou/1a51cf8a95c4875db13b148cd2be2c4121b4514a7765630624c8a4432e61af2a1713d7ec26e18904df7b1d652cebc16829afbcc9b99958cdd64b14c8db397add\\" style=\\"text-decoration: none; color: #297EFF;\\">click here</a><br><br>
  
      You can also follow us on <a href=\\"https://twitter.com/OpenCollect\\" style=\\"text-decoration: none; color: #297EFF;\\">Twitter</a> or come chat with us on our public
      <a href=\\"https://slack.opencollective.com\\" style=\\"text-decoration: none; color: #297EFF;\\">Slack channel</a>.
      <br><br>
      Made with ‚ù§Ô∏è from <a href=\\"https://docs.opencollective.com/help/about/team\\" style=\\"text-decoration: none; color: #297EFF;\\">all over the world</a>
    </td>
  </tr>
  <tr>
    <td colspan=\\"3\\" height=\\"10\\"></td>
  </tr>
  </table><!-- 600px width content table -->
  
  <!--[if (gte mso 9)|(IE)]>
    </td></tr></table>
  <![endif]-->
  
  </td>
  </tr>
  </table><!-- 100% width table -->
  <!-- OpenCollective.com -->
  </body>
  
  </html>"
`;

exports[`server/lib/email Sending emails sends the thankyou.fr email template 2`] = `
"Xavier,

Votre derni√®re contribution de 50,00¬†‚Ç¨ vient d'√™tre re√ßue. Merci pour continuer
√† nous soutenir!

Vous pouvez suivre nos prochaines activit√©s et comment on va utiliser l'argent
recolt√© sur la page de notre collectif ouvert:
https://opencollective.com/enmarchebe [https://opencollective.com/enmarchebe].


AIDEZ-NOUS √Ä R√âCOLTER PLUS D'ARGENT!

Partagez cette URL: https://opencollective.com/enmarchebe/donate
[https://opencollective.com/enmarchebe/donate].

Encore merci! üôè

‚Äì Le collectif En Marche

[http://localhost:3000/static/images/email/logo-email-footer@2x.png]
[http://localhost:3000]

We can do great things together

To unsubscribe from thank you for your donation, click here
[http://localhost:3000/email/unsubscribe/user1%40opencollective.com/enmarchebe/thankyou/1a51cf8a95c4875db13b148cd2be2c4121b4514a7765630624c8a4432e61af2a1713d7ec26e18904df7b1d652cebc16829afbcc9b99958cdd64b14c8db397add]

You can also follow us on Twitter [https://twitter.com/OpenCollect] or come chat
with us on our public Slack channel [https://slack.opencollective.com].

Made with ‚ù§Ô∏è from all over the world
[https://docs.opencollective.com/help/about/team]"
`;

exports[`server/lib/email render Automatically generates text version 1`] = `
"Subject: Open Collective: Login

<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\">
<head>
  <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
  <title></title>
  <style>
@media only screen and (min-device-width: 601px) {
  .content {
    width: 600px !important;
  }
}
.btn.blue:hover {
  background: linear-gradient(180deg,#297EFF 0%,#1869F5 100%);
  background-color: #297EFF;
  border-color: #297EFF;
  color: #FFFFFF;
}
</style>
</head>
<body margin=\\"0\\" padding=\\"0\\" yahoo style=\\"margin: 0; padding: 0; min-width: 100%;\\">

<table bgcolor=\\"white\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\" style=\\"font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;box-sizing:border-box;font-size:14px;\\"><tr><td align=\\"center\\" valign=\\"top\\">
  <!--[if (gte mso 9)|(IE)]>
    <table width=\\"600\\" align=\\"center\\" cellpadding=\\"0\\" cellspacing=\\"0\\" border=\\"0\\"><tr><td>
  <![endif]-->
  <table class=\\"content\\" border=\\"0\\" width=\\"100%\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"width: 100%; max-width: 600px; padding: 0 10px;\\">
    <tr>
      <td></td>
      <td>

<h1 style=\\"color: #3399FF; font-family: 'Helvetica Neue'; font-size: 36px; font-weight: 300; line-height: 44px; text-align: center;\\">üóù</h1>

<center>
  <h3 style=\\"color: #303233; font-family: 'Helvetica Neue'; font-size: 20px; font-weight: bold; line-height: 25px;\\">You are one click away from logging into Open Collective.</h3>
  <a href=\\"https://opencollective/USER_LOGIN_LINK\\" class=\\"btn blue\\" style=\\"text-decoration: none; padding: 10px; border: 2px solid #3399FF; border-radius: 16px; display: inline-block; font-weight: 600; font-size: 14px; line-height: 17px; padding-bottom: 14px; padding-left: 24px; padding-right: 24px; padding-top: 14px; border-radius: 100px; min-height: 18px; background-color: #3399FF; background: linear-gradient(180deg,#1869F5 0%,#1258e2 100%); background-color: #1869F5; border-color: #1869F5; color: #FFFFFF;\\"><div style=\\"font-family: 'Inter', sans-serif; font-weight: 600; font-size: 14px; line-height: 17px; font-size: 12px; text-align: center; color: white;\\">One-click login</div></a>
  <br>
  <br>
  <br>
  <hr>
  <br>
  <p style=\\"color: #494B4D; font-family: 'Helvetica Neue'; font-size: 14px; line-height: 18px;\\">
    Link not working? Try this:<br>
    <a href=\\"https://opencollective/USER_LOGIN_LINK\\" style=\\"text-decoration: none; color: #297EFF; word-break: break-all; font-size: 12px;\\">https://opencollective/USER_LOGIN_LINK</a>
  </p>
</center>

<div itemscope itemtype=\\"http://schema.org/EmailMessage\\">
  <div itemprop=\\"potentialAction\\" itemscope itemtype=\\"http://schema.org/ViewAction\\">
    <link itemprop=\\"url\\" content=\\"https://opencollective/USER_LOGIN_LINK\\">
    <meta itemprop=\\"name\\" content=\\"login\\">
  </div>
  <meta itemprop=\\"description\\" content=\\"Login to opencollective.com\\">
</div>

</td>
</tr>
<tr>
  <td colspan=\\"3\\" height=\\"40\\"></td>
</tr>
<tr>
  <td colspan=\\"3\\" align=\\"center\\">
    <table width=\\"100%\\">
      <tr>
        <td></td>
        <td width=\\"200\\">
          <a href=\\"http://localhost:3000\\" style=\\"text-decoration: none; color: #297EFF;\\">
            <img width=\\"220\\" height=\\"28\\" src=\\"http://localhost:3000/static/images/email/logo-email-footer@2x.png\\" style=\\"max-width: 100%;\\">
          </a>
        </td>
        <td></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
  <td colspan=\\"3\\" align=\\"center\\" style=\\"font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;box-sizing:border-box;font-size:12px;color:#999;\\">
    We can do great things together<br><br>

    You can also follow us on <a href=\\"https://twitter.com/OpenCollect\\" style=\\"text-decoration: none; color: #297EFF;\\">Twitter</a> or come chat with us on our public
    <a href=\\"https://slack.opencollective.com\\" style=\\"text-decoration: none; color: #297EFF;\\">Slack channel</a>.
    <br><br>
    Made with ‚ù§Ô∏è from <a href=\\"https://docs.opencollective.com/help/about/team\\" style=\\"text-decoration: none; color: #297EFF;\\">all over the world</a>
  </td>
</tr>
<tr>
  <td colspan=\\"3\\" height=\\"10\\"></td>
</tr>
</table><!-- 600px width content table -->

<!--[if (gte mso 9)|(IE)]>
  </td></tr></table>
<![endif]-->

</td>
</tr>
</table><!-- 100% width table -->
<!-- OpenCollective.com -->
</body>

</html>
<center>
<div style=\\"margin-top: 1em; font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;box-sizing:border-box;font-size:12px;color:#999;\\">
  This sign in email was requested by
    <span style=\\"padding: .2em .4em; margin: 0; background-color: rgba(27,31,35,.05); border-radius: 3px;\\">
      
    </span>.
    If you did not request this email, please inform <a href=\\"mailto:support@opencollective.com\\" style=\\"text-decoration: none; color: #297EFF;\\">support@opencollective.com</a>.
</div>
</center>
"
`;

exports[`server/lib/email render Automatically generates text version 2`] = `
"üóù


YOU ARE ONE CLICK AWAY FROM LOGGING INTO OPEN COLLECTIVE.

One-click login
[https://opencollective/USER_LOGIN_LINK]




--------------------------------------------------------------------------------


Link not working? Try this:
https://opencollective/USER_LOGIN_LINK [https://opencollective/USER_LOGIN_LINK]



[http://localhost:3000/static/images/email/logo-email-footer@2x.png]
[http://localhost:3000]

We can do great things together

You can also follow us on Twitter [https://twitter.com/OpenCollect] or come chat
with us on our public Slack channel [https://slack.opencollective.com].

Made with ‚ù§Ô∏è from all over the world
[https://docs.opencollective.com/help/about/team]"
`;
